streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala

```diff
@@ -185,7 +185,7 @@ class FileInputDStream[K, V, F <: NewInputFormat[K, V]](
       lastNewFileFindingTime = clock.getTimeMillis()
 
       // Calculate ignore threshold
-      val modTimeIgnoreThreshold = math.max(
+      val modTimeIgnoreThreshold = math.min(
```

Comment:
This looks wrong -- see comment above. Surely it is making sure the threshold is no _earlier_ than either of them?